### PR TITLE
fix(shell): open the shell on your own model when you have no OpenSRE account

### LINKED ISSUE #6112: [BUG] No way to use a local Ollama model in the shell: sign-in is required, and the hosted lock tells you to sign out
## Summary

There is no supported way to run a local Ollama model in the interactive shell. The shell requires an OpenSRE account, a signed-in shell is locked to the hosted OpenAI route, and the lock message tells you to sign out first. Signing out means the shell stops opening at all, so the instruction is a dead end.

Reported on Discord: "How can we configure local ollama models now? I checked Preferences section for models but there was no option to use local models."

## Expected vs actual behavior

Expected: someone with Ollama running can select a local model and use it in the shell.

Actual:

- `opensre` with no arguments hits the sign-in gate. The only two choices are "Sign in or create account" and "Exit and stay signed out" (`surfaces/interactive_shell/ui/sign_in.py:72-99`, wired at `surfaces/interactive_shell/main.py:178`).
- Once signed in, `/model` refuses every change: "LLM settings are managed by your OpenSRE account ... Run `opensre account logout` before configuring a different provider or model" (`surfaces/interactive_shell/command_registry/model/switching.py:16-31`).
- `opensre account logout` clears the account, so the next `opensre` lands back on the gate and the shell never opens.

Our own docs already describe the loop, at `docs/interactive-shell-commands.mdx:565`: "local provider/model changes are locked until sign-out; signing out closes the shell."

The only thing that skips the gate today is `OPENSRE_IS_TEST=1`, `PYTEST_CURRENT_TEST`, or a CI environment (`infrastructure/analytics/source.py:11`). That is not something we can tell a user to set.

## Steps to reproduce

1. `opensre onboard local_llm` and finish the Ollama setup. It succeeds.
2. Run `opensre`. Sign in when the gate asks.
3. Run `/model set ollama llama3.2`. It is refused, and you are told to log out.
4. Run `opensre account logout`.
5. Run `opensre` again. You are back at the sign-in gate with no way into the shell.

## Can you reproduce it consistently?

Yes.

## Scope

Only the interactive shell is affected. `opensre ask`, the gateway, and cron still fall back to the configured `LLM_PROVIDER` when no account route exists (`config/account.py:197`), so Ollama works there.

The web Preferences page showing only hosted models is expected, since the account route is a hosted OpenAI proxy. It is where people look first, though, so whatever we do here needs a line in the docs too.

## Additional context

The mandatory gate is deliberate (#5920), and this is not a request to remove it. What is missing is a supported path for people running their own model. Two options that keep the gate intact:

1. Let the gate pass when a validated local or self-configured provider is already set, so `opensre onboard local_llm` followed by `opensre` works.
2. Add an explicit "use my own model" choice to the sign-in screen, shown only when a provider is configured.

Happy to implement whichever direction the team prefers.

## Environment

macOS 15, `main` at 751e7da2e.

### PR DESCRIPTION
**Pull request by Davidson3556**
Fixes #6112

#### Describe the changes you have made in this PR -

Adds a **Continue with my own model** row to the sign-in gate, so someone running Ollama or their own API key can open the shell without an OpenSRE account. The row appears only when a provider is configured and its credentials resolve, so a fresh install still sees the original two-way choice.

Picking it drops any stored hosted route for the process. That part matters: a session the webapp rejected or could not reach still leaves its record and token on disk, so without dropping it the dead route beats the working local provider. A later `/account login` puts the hosted route back.

Three corrections in the same area:

- The prompt line above the menu said an account was required for the shell. It now matches whatever the menu actually offers.
- The "using your configured provider" notice was printed into a screen `run_repl` clears a moment later, so nobody read it. It moved into the shell, after the banner, and fires only when the gate itself dropped the route. A route already ignored on entry came from the caller's own env var, and telling a signed-in user they are signed out would be wrong.
- `/model` now names `opensre onboard local_llm` when it tells you to log out and you have no provider of your own, since logging out would otherwise close the shell with no way back in.

`README.md` still said the shell "only opens for an active account". Install troubleshooting picks up `OPENSRE_IGNORE_ACCOUNT_ROUTE`, the escape hatch for #6223, where a stale session keeps routing the model on every non-shell surface and `opensre ask` dies with a bare connection error.

### Demo/Screenshot for feature changes and bug fixes -

Signed out with `LLM_PROVIDER` configured. Before, the prompt line contradicted the row under it:

```
Sign in or create an OpenSRE account to use the interactive shell.

  ❯ Sign in or create account
    Continue with my own model
    Exit and stay signed out
```

After:

```
Sign in to OpenSRE, or continue with the model you already configured.

  ❯ Sign in or create account
    Continue with my own model
    Exit and stay signed out
```

Choosing that row, the notice now survives the screen clear and lands under the banner:

```
                        Skills (8) ✓     CI/CD fixes (0)

Signed out. Using your configured LLM provider.

  Ask User
  Which demo would you like me to run?
```

The shell then runs a real turn on the local provider, with no account:

```
$ /model show
LLM connection · local configuration

provider │ reasoning model │ toolcall model
━━━━━━━━━┿━━━━━━━━━━━━━━━━━┿━━━━━━━━━━━━━━━
openai   │ gpt-5.4         │ gpt-5.4

> In one sentence, what does a SIGTERM do to a process?

Ω SIGTERM asks a process to terminate gracefully, giving it a chance to catch
  the signal, clean up, and exit before being forcibly killed.
```

Same flow with a hosted record on disk pointing at an unreachable app URL: the gate still offers the row, and `/model show` reports `local configuration` rather than the dead route. A fresh install (empty `HOME`, no `LLM_PROVIDER`) keeps the two-row gate and the original prompt copy.

Checks run locally: `make lint`, `make format-check`, `make typecheck`, and `uv run python -m pytest tests/config/ tests/interactive_shell/ tests/core/runtime/llm/ tests/quality/ tests/shared/` (2587 passed, 1 skipped).

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

The problem is a dead end that users hit in practice: the shell demands an account, a signed-in shell is pinned to the hosted model, and the `/model` lock tells you to log out, which closes the shell. There was no supported way to reach a local model.

I kept the gate rather than weakening it. The new row is gated on `has_user_configured_llm_provider()`, which requires `LLM_PROVIDER` to be set explicitly and its credentials to resolve. The built-in default does not count, so the row cannot appear on a fresh install and become a way around sign-in.

The alternative I considered was letting the gate auto-pass whenever a local provider exists. I did not do that, because it silently removes the gate for anyone with a stray `LLM_PROVIDER` in their environment. Making it an explicit choice keeps the decision with the user.

Dropping the hosted route needed process-wide state, since `account_llm_route()` is consulted from the LLM factory and the cache key, not just the gate. `OPENSRE_IGNORE_ACCOUNT_ROUTE` does that, and because it feeds `current_llm_client_cache_key()`, cached clients rebuild against the local provider instead of being served stale.

For the notice, I first keyed it on the env var, then changed it: a user who exports that variable themselves is not making a choice in the gate, and could be signed in, so the message would have lied. `run_repl` now compares the route state either side of the gate and only announces a flip it caused.

Key pieces: `run_sign_in_gate` owns the choice loop and stays injection-based; `account_llm_route_ignored()` is the predicate both the route resolver and the shell read; `render_own_model_notice()` is shared by the gate path and `/account logout` so the wording cannot drift.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [ ] I have performed a self-review of my code
- [ ] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

### CODE REVIEW

File: config/llm_settings.py [hunk 1]
```diff
@@ -487,3 +488,25 @@ def has_credentials_for_active_llm_provider() -> bool:
...
+        return False
+    try:
+        settings = resolve_llm_settings()
+    except Exception:
+        # Credentials alone do not make a provider usable: azure-openai needs an
+        # endpoint, vertex-ai a project. Resolving here keeps callers from
+        # offering a route that fails on the first turn.
+        return False
+    auth_status = credential_status(settings.provider)
+    return auth_status.configured and not auth_status.stale
```
[the remark this answers is not in the document]
Comment (Davidson3556) [reply config/llm_settings.py]: Good catch, fixed in a02388d. The gate now confirms the credential resolves through resolve_for_request rather than trusting the record, so a provider whose secret is gone no longer gets offered. Keyless kinds return ok without touching storage, so the Ollama path is unchanged and startup does not read secrets for it.

File: config/account.py [hunk 2]
```diff
@@ -194,8 +195,31 @@ def delete_account_token() -> None:
...
+
+
+def restore_account_llm_route() -> None:
+    """Re-enable the hosted route after a login replaces the rejected session."""
+    os.environ.pop(OPENSRE_IGNORE_ACCOUNT_ROUTE_ENV, None)
+
+
+def account_llm_route_ignored() -> bool:
+    """Whether this process dropped the hosted route in favour of the user's own model."""
+    return bool(os.getenv(OPENSRE_IGNORE_ACCOUNT_ROUTE_ENV, "").strip())
```
[the remark this answers is not in the document]
Comment (Davidson3556) [reply config/account.py]: Agreed, fixed in a02388d. It now parses against the same false set the rest of the repo uses ("", 0, false, off, no). This matters more than it looks because the same PR documents the variable in install troubleshooting, so users will start setting it by hand.

**Comment by Davidson3556** [comment]
@greptile review

**Comment by muddlebee** [comment]
@Davidson3556 we have discussed this earlier, this is not the right approach. 

Please wait for clarity from core team before introducing a new feature line in CLI.
